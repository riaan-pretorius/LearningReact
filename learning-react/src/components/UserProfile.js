// @flow
import React, { Component } from "react";
import { Col, Card, CardBlock, CardTitle, CardSubtitle } from "reactstrap";
import axios from "axios";

type Props = {
  username: string
};
type State = {
  username: string,
  user: ?Object,
  loading: boolean
};

class UserProfile extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      loading: false,
      username: props.username,
      user: null
    };
    //this.getUser(props.username);
  }
  componentDidMount() {
    this.getUser(this.state.username);
  }
  componentWillReceiveProps(newProps) {
    this.getUser(newProps.username);
  }
  getUser = (username: string) => {
    if (this.state.username !== username) {
      this.setState({ loading: true });
      axios
        .get("https://api.github.com/users/" + username)
        .then(response => {
          this.setState({
            username: username,
            user: response.data,
            loading: false
          });
        })
        .catch(function() {
          console.log("Promise Rejected");
        });
    }
  };
  render() {
    return (
      <Col>
        {this.state.loading || !this.state.user ? (
          "Loading..."
        ) : (
          <Col>
            <Card>
              <CardBlock>
                <CardTitle>
                  <strong>Username:</strong>
                  {this.state.user.login}
                </CardTitle>
                <CardSubtitle>
                  <strong>Name:</strong>
                  {this.state.user.name}
                </CardSubtitle>
              </CardBlock>
              <img
                src={this.state.user.avatar_url}
                alt="Generic placeholder"
                style={styles.profileImage}
              />
            </Card>
          </Col>
        )}
      </Col>
    );
  }
}

const styles = {
  profileImage: {
    borderStyle: "outset"
  }
};

export default UserProfile;
