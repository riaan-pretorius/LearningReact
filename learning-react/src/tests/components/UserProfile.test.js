import React from "react";
import UserProfile from "../../components/UserProfile";
import { mount, shallow } from "enzyme";
import renderer from "react-test-renderer";

test("UserProfileTest Snapshot", () => {
  const component = renderer.create(<UserProfile username="riaan-pretorius" />);
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});

test("UserProfile Test", () => {
  const component = mount(<UserProfile username="riaan-pretorius" />);
  setTimeout(() => {
    expect(component.find(".card").exists()).toBeTruthy();

    let header = component.find("h4");
    expect(header.exists()).toBeTruthy();
    expect(header.text()).toBe("Username:riaan-pretorius");

    let p = component.find("h6");
    expect(p.exists()).toBeTruthy();
    expect(p.text()).toContain("Name:");

    let button = component.find("img");
    expect(button.exists()).toBeTruthy();
  }, 2000);
});
