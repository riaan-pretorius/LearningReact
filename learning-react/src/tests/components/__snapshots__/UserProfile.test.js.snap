// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileTest Snapshot 1`] = `
<div
  className="col"
>
  Loading...
</div>
`;
